File: testsuite/cppkeepbodies/tifdef_in_class.cpp

#assumedef windows

class failClass1 {
public:
  int tmpI;
#ifdef windows
  void someProc(int x);
#else
  void* myfield;
#endif

#if 0
  int atLeastSkipThis;
#endif
};

class failClass2 {
public:
  int tmpI;
  #define myconst 122
#ifdef unknown
  void someIfdefProc(int x);
#else
  void* myfield;
#endif
};

File: testsuite/results/tifdef_in_class.nim
type
  failClass1* {.bycopy.} = object
    tmpI*: cint
    when 0:
      atLeastSkipThis*: cint


proc someProc*(this: var failClass1; x: cint)
type
  failClass2* {.bycopy.} = object
    tmpI*: cint
    when not defined(unknown):
      myfield*: pointer


const
  myconst* = 122

when defined(unknown):
  proc someIfdefProc*(this: var failClass2; x: cint)